src/OpenSage.Game/Logic/Object/Body/ActiveBody.cs

```diff
@@ -110,9 +110,15 @@ internal override void Load(StatePersister reader)
 
             var maxHealth2 = _maxHealth;
             reader.PersistSingle(ref maxHealth2);
-            if (_maxHealth != maxHealth2)
+
+            if (reader.SageGame == SageGame.CncGeneralsZeroHour)
```

Comment:
for behavior like this, when I did it I went with `if (reader.SageGame is not SageGame.CncGenerals)` since I assumed all future games would use this new version - but I also didn't verify.

Reply:
Good catch - https://github.com/OpenSAGE/OpenSAGE/commit/b54a4fc795ff4f83fb2043871ff4e6747b1a56d2